provider/datagen/src/baked_exporter.rs

```diff
@@ -44,30 +44,30 @@
 //! use icu_locid::langid;
 //! use icu_provider::hello_world::*;
 //!
-//! pub struct MyDataProvider;
+//! # macro_rules! include {
+//! #   ($path:literal) => {}
+//! # }
+//! # macro_rules! impl_data_provider {
+//! #   ($p:ty) => {
+//! #     use icu_provider::prelude::*;
+//! #     use icu_provider::hello_world::*;
+//! #     impl DataProvider<HelloWorldV1Marker> for $p {
+//! #       fn load(&self, req: DataRequest) -> Result<DataResponse<HelloWorldV1Marker>, DataError> {
+//! #         HelloWorldProvider.load(req)
+//! #       }
+//! #     }
+//! #   }
+//! # }
+//! include!("/tmp/icu4x_baked_demo/mod.rs");
 //!
-//! const _: () = {
```

Comment:
update because the `const _: ()` approach doesn't work if there are recursive `include!`s.